[WIP] roachtest: add tenant to tenant streaming roachtest

adityamaru opened the pull request:
Release note: None

adityamaru committed:
[WIP] roachtest: add tenant to tenant streaming roachtest

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62190)
<!-- Reviewable:end -->

adityamaru committed:
roachprod: add --new-kv-server option to start a new cluster

Previously, roachprod would not allow starting multiple clusters across
VMs.  Node 1 was special cased for cluster initialization and `roachprod
start` on every other node would implicitly join node 1, thereby
creating a single cluster.

BulkIO has been working on a tenant to tenant streaming feature that
requires a source KV server and a destination KV server, so that one can
stream a tenant from the source to the destination. Currently, such a
setup is not possible with roachprod making it difficult to write
roachtests to this effect. I suspect such use cases will increasingly
come up as we build ou multi-tenancy.

This change adds a `--new-kv-server` flag that can be specified during
`roachprod start`. The node being started with this flag will be treated
as "node 1" of the new cluster. i.e. it will join itself and perform
cluster intialization.  Subsequent `roachprod start`s can choose which
cluster to join by supplying the `--join` arguments.

This does not change the default behavior of roachprod.

Release note: None
